hydra-node/json-schemas/api.yaml

```diff
@@ -620,285 +391,580 @@ components:
+        - seq
+        - timestamp
+      properties:
+        tag:
+          type: string
+          enum: ["HeadIsClosed"]
+        headId:
+          $ref: "api.yaml#/components/schemas/HeadId"
+        snapshotNumber:
+          $ref: "api.yaml#/components/schemas/SnapshotNumber"
+        contestationDeadline:
+          $ref: "api.yaml#/components/schemas/UTCTime"
+        seq:
+          $ref: "api.yaml#/components/schemas/SequenceNumber"
+        timestamp:
+          $ref: "api.yaml#/components/schemas/UTCTime"
+
+    HeadIsContested:
+      type: object
+      required:
+        - tag
+        - headId
+        - snapshotNumber
+        - seq
+        - timestamp
+      properties:
+        tag:
+          type: string
+          enum: ["HeadIsContested"]
+        headId:
+          $ref: "api.yaml#/components/schemas/HeadId"
+        snapshotNumber:
+          $ref: "api.yaml#/components/schemas/SnapshotNumber"
+        seq:
+          $ref: "api.yaml#/components/schemas/SequenceNumber"
+        timestamp:
+          $ref: "api.yaml#/components/schemas/UTCTime"
+
+    ReadyToFanout:
+      type: object
+      required:
+        - tag
+        - headId
+        - seq
+        - timestamp
+      properties:
+        tag:
+          type: string
+          enum: ["ReadyToFanout"]
+        headId:
+          $ref: "api.yaml#/components/schemas/HeadId"
+        seq:
+          $ref: "api.yaml#/components/schemas/SequenceNumber"
+        timestamp:
+          $ref: "api.yaml#/components/schemas/UTCTime"
+
+    HeadIsAborted:
+      type: object
+      required:
+        - tag
+        - headId
+        - utxo
+        - seq
+        - timestamp
+      properties:
+        tag:
+          type: string
+          enum: ["HeadIsAborted"]
+        headId:
+          $ref: "api.yaml#/components/schemas/HeadId"
+        utxo:
+          $ref: "api.yaml#/components/schemas/UTxO"
+        seq:
+          $ref: "api.yaml#/components/schemas/SequenceNumber"
+        timestamp:
+          $ref: "api.yaml#/components/schemas/UTCTime"
+
+    HeadIsFinalized:
+      type: object
+      required:
+        - tag
+        - headId
+        - utxo
+        - seq
+        - timestamp
+      properties:
+        tag:
+          type: string
+          enum: ["HeadIsFinalized"]
+        headId:
+          $ref: "api.yaml#/components/schemas/HeadId"
+        utxo:
+          $ref: "api.yaml#/components/schemas/UTxO"
+        seq:
+          $ref: "api.yaml#/components/schemas/SequenceNumber"
+        timestamp:
+          $ref: "api.yaml#/components/schemas/UTCTime"
+
+    TxValid:
+      type: object
+      required:
+        - tag
+        - headId
+        - transaction
+        - seq
+        - timestamp
+      properties:
+        tag:
+          type: string
+          enum: ["TxValid"]
+        headId:
+          $ref: "api.yaml#/components/schemas/HeadId"
+        transaction:
+          $ref: "api.yaml#/components/schemas/Transaction"
```

Comment:
Must: make this compatible by `cardano-cli`.

As we have detailed in the issue description over the last couple of days/weeks, we care about compatibility of this output with `cardano-cli`. That means, that whatever we get back here, it should be understandable by `cardano-cli`, for example to display a human readable rendering of the transaction contents.

However, when I do try it, it fails on a missing `description` field:

```
websocat -U ws://127.0.0.1:4001 | jq 'select(.tag == "TxValid") | .transaction' | cardano-cli transaction view --tx-file /dev/stdin
```
```
Command failed: transaction view  Error: Failed to decode neither the cli's serialisation format nor the ledger's CDDL serialisation format. TextEnvelope error: /dev/stdin: TextEnvelope aeson decode error: Error in $: key "description" not found
TextEnvelopeCddl error: /dev/stdin: Could not JSON decode TextEnvelopeCddl file at: /dev/stdin Error: Error in $: key "description" not found
```